### PR TITLE
CONSOLE-3666: Validate shared modules of dynamic plugins

### PR DESCRIPTION
**Pull request by vojtechszocs**
### Changes in Console dynamic plugin SDK
- add optional `options` argument to `ConsoleRemotePlugin` constructor
  - control JSON schema validation: `validatePackageSchema`, `validateExtensionSchema`
  - control extension integrity validation (via `ExtensionValidator`): `validateExtensionIntegrity`
  - :star: control consumed shared module validation: `validateSharedModules`

:star: This functionality is new, other options just allow controlling the existing built-in functionality.

### Changes in Console dynamic demo plugin
- update `react-router` and `react-router-dom` dependencies to Console provided semver range
- update typing dependencies for `react-router` and `react-router-dom`
- remove unused dependencies `comment-json` and `read-pkg`

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts [hunk 1]
```diff
@@ -35,22 +38,98 @@ export const validatePackageFileSchema = (
...
+      );
+    }
+  });
+
+  return result;
+};
+
+export type ConsoleRemotePluginOptions = Partial<{
+  /**
+   * Validate the plugin's `package.json` file schema?
```
Comment (spadgett) [inline frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts:71]: I would add some more detailed comments about what each of these means. (Presumably this shows up in the hover in VSCode?)

Let's add this to the README as well.

**Review by spadgett — APPROVED** [review]
Thanks, @vojtechszocs 👍  LGTM, let's make sure this is well documented.

**Comment by vojtechszocs** [comment]
Requesting approvals:

QE Approver: 
/assign @yanpzhan
Docs Approver:
 /assign @ahardin-rh
PX Approver:
/assign @RickJWagner

This PR adds an additional type of validation when building Console dynamic plugins.

No expected impact on existing plugins, assuming their shared module dependencies are consistent with what is provided by target Console version (based on `@openshift-console/dynamic-plugin-sdk` package version used by the plugin).

**Review by opayne1 — COMMENTED** [review]
/label docs-approved

**Comment by vojtechszocs** [comment]
Fixed merge conflict in `dynamic-demo-plugin/yarn.lock` file.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**